src/jit.rs

```diff
@@ -1554,7 +1556,7 @@ impl<'a, C: ContextObject> JitCompiler<'a, C> {
                 };
                 self.emit_rust_call(Value::Constant64(store, false), &[
                     Argument { index: 3, value: Value::Register(REGISTER_SCRATCH) }, // Specify first as the src register could be overwritten by other arguments
-                    Argument { index: 2, value: Value::Register(REGISTER_OTHER_SCRATCH) },
+                    Argument { index: 2, value: Value::RegisterIndirect(RSP, -8, false) },
```

Comment:
Shouldn't the offset for `stack_slot_of_value_to_store ` (i.e. -112) be the same as the one here (-8)?

Reply:
At this point `emit_rust_call` has already saved all the VM register state, thus the stack is taller. RSP-8 is pointing to the next stack slot to be pushed. This is where we temporarily saved the "value to store" at. It will be overwritten by the `RIP` in the subsequent call.